src/interaction/encode.ts

```diff
@@ -52,14 +62,15 @@ export default async function(type: Event): Promise<void> {
             unload.reset();
             queue(tokens);
             break;
-        case Event.InputChange:
-            let ch = change.data;
-            if (ch) {
-                tokens.push(observe(ch.target as TargetInfo));
-                tokens.push(ch.value);
-                change.reset();
+        case Event.Input:
+            for (let i = 0; i < input.state.length; i++) {
+                let entry = input.state[i];
```

Comment:
Why are you iterating over a list of things for the state here?

Reply:
We look ahead for 500ms to receive input. If we get input events for two fields (someone typing fast, etc.) then we will queue up two different input events (one for each individual target). This loop helps us ensure we send all events.